framework/components/ecl-carousels/carousels.js

```diff
@@ -0,0 +1,175 @@
+import {
+  addClass,
+  removeClass,
+  queryAll,
+} from '@ec-europa/ecl-base/helpers/dom';
+
+import debounce from 'lodash.debounce';
+
+/**
+ * @param {object} options Object containing configuration overrides
+ */
+export const carousels = ({ selectorId: selectorId = 'ecl-carousel' } = {}) => {
+  // SUPPORTS
+  if (!('querySelector' in document) || !('addEventListener' in window)) {
+    return null;
+  }
+
+  // SETUP
+  let currentSlide = 0;
+  const carousel = document.getElementById(selectorId);
+  const slides = queryAll('.ecl-carousel__item', carousel);
+  const list = carousel.querySelector('.ecl-carousel__list');
+
+  function getListItemWidth() {
+    return carousel.querySelector('.ecl-carousel__item').offsetWidth;
+  }
+
+  function alignCenter() {
+    slides.forEach(slide => {
+      // eslint-disable-next-line
+      slide.style.left = `${(list.offsetWidth - slide.offsetWidth) / 2}px`;
+    });
+  }
+
+  function goToSlide(n) {
+    removeClass(slides[currentSlide], 'ecl-carousel__item--showing');
+    currentSlide = (n + slides.length) % slides.length;
+    addClass(slides[currentSlide], 'ecl-carousel__item--showing');
+  }
+
+  function setOffset() {
+    const itemWidth = getListItemWidth();
+    const tr = `translate3d(${-currentSlide * itemWidth}px, 0, 0)`;
+
+    list.style.MozTransform = tr; /* FF */
+    list.style.msTransform = tr; /* IE (9+) */
+    list.style.OTransform = tr; /* Opera */
+    list.style.WebkitTransform = tr; /* Safari + Chrome */
+    list.style.transform = tr;
+  }
+
+  function announceCurrentSlide() {
+    carousel.querySelector(
+      '.ecl-carousel__meta-slide'
+    ).textContent = `${currentSlide + 1} / ${slides.length}`;
+  }
+
+  function showImageInformation() {
+    // Reset/Hide all.
+    const infoAreas = queryAll('[data-image]');
+    // If anything is visible.
+    if (infoAreas) {
+      // eslint-disable-next-line
+      infoAreas.forEach(area => (area.style.display = "none"));
```

Comment:
The logic at the moment is different for 2 reasons, but I can change if necessary:
- image hide/show is with opacity + translation since if there is no opacity 0 on hidden items, you will actually see the images next to each other like a series of frames. Another technique, of course, is to have items absolutely positioned in a relative container and display block only the featured. I personally decided to keep the series of frame like way so that if some day an animation is needed, it will definitely look much better
- also I decided to use js for part of the css transformation as they don't make sense in no-js mode

I am open for feedback in overall